Home work23b

TestProGio opened the pull request:
I used the same exercise for hw22, since I would like to figure out why its running for you but not me. Still trying. Looking forward to your feedback.

Giovanna

TestProGio committed:
First Commit on my Fork

Forked a cloned repo from A59 testPro

TestProGio committed:
HW 19

. Create a new branch named homework-19.

2. Create a new file Homework19.java in IntelliJ IDEA.

3. Create a test case, deletePlaylist() using @Test annotation.

4. Use the helper/reusable methods we created.

5. Use @Parameters for passing baseUrl from the TestNG config file to the tests.

6. Navigate to "https://qa.koel.app/".

7. Log in with your credentials.

8. Click the playlist you want to delete.

9. We should see a red button "x PLAYLIST" on the top right part of the page, and click on it.

Note: If the red button is not displayed, let's create a new playlist and click it.

10. Verify that the confirmation notification displayed has the text, "Deleted playlist {playlist name}".

11. Commit your changes to the new branch homework-19.

12. Push your code to the remote repository.

13. Create a pull request.

TestProGio committed:
Homework-19 v2

I included 'Assert.assertTrue(getSuccessPopUp().isDisplayed());' with your helpful tips. Thanks. Gio

TestProGio committed:
Homework 22

I need your guidance on making this run. Focusing of recreating HW19 and creating PlaylistActions. java and PlaylistActionsTest. Please guide me. Everything looks like it should run.

TestProGio committed:
homework22-b

Implementing your feedback, but still have issues on my side.

TestProGio committed:
homework-23

Berichicko commented:
You got this problem, please correct it.

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/b567c7ca-e285-4f69-9454-8da891dd4df8)

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/b2f5f295-88fe-4403-ad64-7fcd861c2867)


**HINT**

Step 1

**Use these methods in the BasePage, and then in child classes**

For example

```
package pgFactoryPages;

import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.interactions.Actions;

import java.time.Duration;

public class BasePage {
    //HW Feedback: Use 'protected' access since it gives the subclass a chance to use the helper method or variable, while preventing a non-related class from trying to use it.
    protected WebDriver driver;
    protected WebDriverWait wait;
    protected Actions actions;

    By soundBarVisualizer = By.cssSelector("[data-testid = 'sound-bar-play']");

    public BasePage(WebDriver givenDriver){
        driver = givenDriver;
        wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        actions = new Actions(driver);
        PageFactory.initElements(driver, this);
    }

    // Waite Methods
    public WebElement waitForPresence(WebElement element) {
        return wait.until(ExpectedConditions.visibilityOf(element));
    }

    public List<WebElement> waitForPresenceOfAllElements(List<WebElement> elements) {
        return wait.until(ExpectedConditions.visibilityOfAllElements(elements));
    }

    public WebElement waitForVisibility(WebElement element) {
        return wait.until(ExpectedConditions.visibilityOf(element));
    }

    public List<WebElement> waitUntilAllElementsLocatedByAreVisible(List<WebElement> elements) {
        return wait.until(ExpectedConditions.visibilityOfAllElements(elements));
    }

    public boolean waitUntilRequiredElementsAreInvisible(List<WebElement> elements) {
        return wait.until(ExpectedConditions.invisibilityOfAllElements(elements));
    }

    public WebElement waitForClickable(WebElement element) {
        return wait.until(ExpectedConditions.elementToBeClickable(element));
    }

    // Click Methods
    public void click(WebElement element) {
        waitForClickable(element).click();
    }

    public void waitForClickableAndClick(WebElement element) {
        WebElement clickableElement = waitForClickable(element);
        clickableElement.click();
    }

    public void tryClick(WebElement element) {
        waitForVisibility(element).click();
    }

    public void tryClickNoWait(WebElement element) {
        element.click();
    }

    public void doubleClick(WebElement element) {
        WebElement clickableElement = waitForClickable(element);
        actions.doubleClick(clickableElement).perform();
    }

    public void mouseOverAndClick(WebElement element) {
        actions.moveToElement(element).click().perform();
    }

    public void mouseOverAndClickAndHold(WebElement element) {
        actions.moveToElement(element).clickAndHold().perform();
    }
}
```

**Step 2**

**PlaylistActions extends BasePage**

For example

```
package pgFactoryPages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class PlaylistActions extends BasePage {


    // Constructor
    public PlaylistActions(WebDriver givenDriver) {
        super(givenDriver);
    }

    // LOCATORS
    @FindBy(linkText = "Delete2Sample Edited PlaylistSample Edited PlaylistSample Edited PlaylistSample Edited Playlist")
    private WebElement clickPlaylist;

    @FindBy(css = "button[title='Delete this playlist']")
    private WebElement deletePlaylistAction;

    @FindBy(css = "div.success.show")
    private WebElement successPopUp;

    // Page Methods
    public void selectPlayList() {
        waitForPresence(clickPlaylist).click();
        System.out.println("Clicked Playlist");
    }

    public void deletePlaylist() {
        waitForClickable(deletePlaylistAction);
        System.out.println("Deleted Playlist");
    }

    public void successPopUpBox() {
        waitForPresence(successPopUp);
        System.out.println("Success Popup Displayed");
    }
}
```

**Step 3**

**1**

**Select File | Invalidate Caches from the main menu:**


![image](https://github.com/Test-Pro-Bootcamp/A55-QA-Automation-Framework/assets/104761482/334be776-4810-40d1-98cd-ad3b02f2b753)


**2**

**In the Invalidate Caches dialog, you can select :**

- Clear file system cache and Local History: remove the virtual file system cache together with the information stored in Local History.This action might be helpful for troubleshooting purposes when the usual cache invalidation is not enough to solve the problem.

- Clear VCS Log caches and indexes: remove the cache and indexes of the VCS Log.Enable this option if you are certain that your VCS Log indexes are broken, for example, if the Log search doesn't work correctly.

- Mark downloaded shared indexes as broken: do not use the already downloaded shared indexes. After invalidation, IntelliJ IDEA will download and use new indexes if they are available. Otherwise, new indexes will be built locally.

![image](https://github.com/Test-Pro-Bootcamp/A55-QA-Automation-Framework/assets/104761482/e41a2dc7-0c52-4d1b-9929-f4fe2c663669)

TestProGio committed:
homework23 updated

Applied your changes but a bit confused on this change to 'PlaylistActions':   // LOCATORS
    @FindBy(linkText = "Delete2Sample Edited PlaylistSample Edited PlaylistSample Edited PlaylistSample Edited Playlist")
    private WebElement clickPlaylist;

TestProGio committed:
HW 23

Very challenging since it runs and then it doesn't but here it is.
